File: mirage/factories/place.js
import { Factory, faker } from 'ember-cli-mirage';

export default Factory.extend({
  name() {
    return faker.address.city();
  },
  slug() {
    return faker.helpers.slugify(this.name);
  },
  confidence: "3",
  // source(){
  //   return "GeoNames";
  // },
  // geonameid(){
  //   return faker.lorem.word();
  // },
  // what3word(){
  //   return faker.lorem.words();
  // },
  note() {
    return faker.lorem.paragraph();
  },
  nickname_sentence() {
    return faker.lorem.sentence();
  },
  lat() {
    return faker.address.latitude();
  },
  lon() {
    return faker.address.longitude();
  },
  instance_count() {
    return faker.random.number();
  },
  flagged: false
});

File: spec/factories/place.rb
# frozen_string_literal: true
FactoryBot.define do
  factory :place do
    name { Faker::Address.city }
    lat { Faker::Address.latitude }
    lon { Faker::Address.longitude }
    confidence "3"
    source "Geonames"
    user
  end

  factory :no_place, class: Place do
    name { Faker::Address.city }
    lat { Faker::Address.latitude }
    lon { Faker::Address.longitude }
    confidence "0"
    source "Geonames"
    user
  end

  factory :fake_place, class: Place do
    name { Faker::Address.city }
  end

  factory :edit_place, class: Place do
    lat { 10 }
    lon { 10 }
  end

end
